cpp/src/arrow/array/builder_primitive.h

```diff
@@ -32,6 +32,16 @@ class ARROW_EXPORT NullBuilder : public ArrayBuilder {
   explicit NullBuilder(MemoryPool* pool ARROW_MEMORY_POOL_DEFAULT)
       : ArrayBuilder(null(), pool) {}
 
+  /// \brief Append the specified number of null elements
+  Status AppendNulls(int64_t length) {
+    auto new_length = length_ + length;
+    ARROW_RETURN_NOT_OK(CheckCapacity(new_length, length_));
```

Comment:
Could you also add this check to existing `AppendNull()`?
Other builders check this. So `AppendNull()` should also check this.